core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -213,6 +213,11 @@ oppia.factory('oppiaPlayerService', [
                 _explorationId));
           }
           loadedExploration.then(function(data) {
+            // Update exploration's states field.
+            for (var stateName in data.exploration.states) {
```

Comment:
No, don't do this. You'll need to modify the functions explicitly, e.g. ExplorationObjectFactory.createFromBackendDict(data.exploration, data.state_classifier_mapping). Be careful to check if there are other instances of ExplorationObjectFactory.createFromBackendDict() that don't have access to this data -- if so, you might need to construct the state stuff in the backend instead and then pass it over.